agent/uiserver/uiserver.go

```diff
@@ -92,20 +92,20 @@ func (h *Handler) ReloadConfig(newCfg *config.RuntimeConfig) error {
 func (h *Handler) handleIndex() (http.Handler, error) {
 	cfg := h.getRuntimeConfig()
 
-	var fsys http.FileSystem
+	var fsys fs.FS
 	if cfg.UIConfig.Dir == "" {
 		// strip the dist/ prefix
-		sub, err := fs.Sub(assets, "dist")
+		sub, err := fs.Sub(dist, "dist")
 		if err != nil {
 			return nil, err
 		}
-		fsys = http.FS(sub)
+		fsys = sub
 	} else {
-		fsys = http.Dir(cfg.UIConfig.Dir)
+		fsys = os.DirFS(cfg.UIConfig.Dir)
```

Comment:
Effectively the same behaviour, relying on `os.Open`:
[`http.Dir`](https://cs.opensource.google/go/go/+/refs/tags/go1.16.7:src/net/http/fs.go;l=71) vs [`os.DirFS`](https://cs.opensource.google/go/go/+/refs/tags/go1.16.7:src/os/file.go;l=636)